lib/ui/key.dart

```diff
@@ -27,6 +67,7 @@ class KeyData {
     required this.logical,
     required this.character,
     required this.synthesized,
+    this.deviceType = KeyEventDeviceType.keyboard,
```

Comment:
In order to pass the smoke test for the framework, this needs to be optional.  Should we keep it that way, just to make migration easier? I'm leaning towards "yes", although all the other properties are required.

Reply:
Yeah I think it's fine. At the end of the day it's `PlatformDispatcher` that creates this object, right?

Reply:
Yes, usually, although you could write your own test framework or something that used it.